src/main/java/co/cask/hydrator/salesforce/plugin/source/batch/util/SalesforceSourceConstants.java

```diff
@@ -32,4 +31,6 @@
   public static final int DURATION_DEFAULT = 0;
   public static final int OFFSET_DEFAULT = 0;
   public static final int INTERVAL_FILTER_MIN_VALUE = 0;
+  public static final int SOQL_MAX_LENGTH = 20000;
```

Comment:
are these values determined by salesforce?

Reply:
Yes, both constants `SOQL_MAX_LENGTH` and `WIDE_QUERY_MAX_BATCH_COUNT` are determined by Salesforce limitations. 
https://developer.salesforce.com/docs/atlas.en-us.salesforce_app_limits_cheatsheet.meta/salesforce_app_limits_cheatsheet/salesforce_app_limits_platform_soslsoql.htm
https://developer.salesforce.com/docs/atlas.en-us.api.meta/api/sforce_api_calls_retrieve.htm